packages/hardhat-core/test/internal/hardhat-network/provider/utils/runFullBlock.ts

```diff
@@ -69,57 +67,27 @@ export async function runFullBlock(
     }
   );
 
-  // TODO uncomment and fix this
-  // forkedNode["_vmTracer"].disableTracing();
-
-  const afterBlockEvent = await runBlockAndGetAfterBlockEvent(
-    // TODO remove "as any" and make this work with VMAdapter
-    forkedNode["_vm"] as any,
-    {
-      block,
-      generate: true,
-      skipBlockValidation: true,
-    }
-  );
-
-  const modifiedBlock = afterBlockEvent.block;
-
-  // TODO remove "as any" and make this work with VMAdapter
-  await (forkedNode["_vm"] as any).blockchain.putBlock(modifiedBlock);
-  await (forkedNode["_vm"] as any).putBlock(modifiedBlock);
-  await forkedNode["_saveBlockAsSuccessfullyRun"](
-    modifiedBlock,
-    afterBlockEvent as any // TODO remove this as any
-  );
-
-  const newBlock = await forkedNode.getBlockByNumber(blockToRun);
-
-  if (newBlock === undefined) {
-    assert.fail();
-  }
-
-  await assertEqualBlocks(newBlock, afterBlockEvent, rpcBlock, forkClient);
-}
-
-async function runBlockAndGetAfterBlockEvent(
-  vm: VM,
-  runBlockOpts: RunBlockOpts
-): Promise<AfterBlockEvent> {
-  let results: AfterBlockEvent;
-
-  function handler(event: AfterBlockEvent) {
-    results = event;
+  const vm = forkedNode["_vm"];
+
+  const blockBuilder = new BlockBuilder(vm, common, {
+    parentBlock,
+    headerData: block.header,
+  });
+  await blockBuilder.startBlock();
+
+  for (const [i, tx] of block.transactions.entries()) {
+    console.log(
```

Comment:
Still has `console.log`

Reply:
This was originally done by Gene. I just commented that he left a `console.log` entry. I'll just remove it myself 😁